### DISCUSSION TITLE
TP4 - Ejercicio 8 - Busqueda

### DISCUSSION THREAD

**Discussion by Elias-Turner** [INGCOM-UNRN-P1/cursada-2023#138] [Preguntas y respuestas] [answered]
Implementar una función que retorna el puntero relativo a una cadena segura, en donde se encuentre la primera ocurrencia de una segunda.

no entiendo la parte de puntero relativo y la primera ocurrencia

**Comment by TeoInMaghine** [comment]
> puntero relativo a una cadena segura

Significa que es un puntero que apunta a algún carácter dentro de la cadena, que podés calcular como el puntero de esa cadena + el índice del carácter.

> primera ocurrencia

Se refiere a la primera vez que aparece la cadena que estás buscando.

Entonces lo que te pide es que devuelvas el puntero de la primera vez que encuentra la cadena buscada, en la cadena donde se hace la búsqueda.

**Comment by martinvilu** [comment, marked as answer]
De la consigna;
> Implementar una función que retorne el puntero relativo a una cadena segura

La idea es usar aritmetica de punteros para retornar la dirección de donde esta lo que
estas buscando, relativo a donde lo estas buscando


Cadena: `Hola Mundo Programación`
Búsqueda: `Mundo`
Puntero a la `M` en cadena

En un ejemplo un poco mas complejo,
Cadena: `El perro marron come carbon en una tarde de con coscorron`
Búsqueda: `on`
Retorno, el primer `on` en la cadena, en la 13 ava posición

**Comment by martinvilu** [comment]
Duplicado (o casi) de https://github.com/INGCOM-UNRN-P1/cursada-2023/discussions/142
